packages/editor/src/diagram/snap.ts

```diff
@@ -9,4 +9,15 @@ export class IvyGridSnapper extends GridSnapper {
   constructor() {
     super({ x: IvyGridSnapper.GRID_X, y: IvyGridSnapper.GRID_Y });
   }
+
+  snap(delta: Point, _element: GModelElement): Point {
+    return onGrid(delta, this.grid);
+  }
+}
+
+export function onGrid(point: Point, grid: Point): Point {
+  return {
+    x: Math.trunc(point.x / grid.x) * grid.x,
```

Comment:
The only change to the original is that here we use `trunc` instead of `round`? In my opinion this is not necessary as round looks better to me if I look at the failing test cases.
So in my opinion this code can be removed.

Reply:
There are some changes in the feedback, especially the ghost element feedback as well to make sure we properly snap the mouse position first, regard that as the center and then calculate the position from there.

I used trunc here because I thought it was closer to the server implementation where we simply cut away the decimals but I missed updating the test cases. I can also change it back to `round` again, as you wish.